### ISSUE TITLE
RFID-token exchange: UID, eMA-id and optional 'printed card number'

### ISSUE THREAD

**Issue by rutgerplantenga** [ocpi/ocpi#31]
There are a number of known cases where the eMA-id cannot be printed onto the RFID card, for instance because it's a full-service mobility card that's already covered with multiple numbers. 

To still facilitate the communication about the card (customer calling operator helpdesk, giving them his/her card-number), it would be good if also a 'printed card number' can be exchanged as an optional field. 

This should  probably be an alphanumeric field as it is not necessarily just a number.

**Comment by simonschllng** [comment]
In OCHP we use:

| Field Name    | Field Type   | Card. | Description                                                                                                                                                                    |
|---------------|--------------|-------|--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
| EmtId         | EmtId        | 1     | Electrical Vehicle Contract Identifier                                                                                                                                         |
| contractId    | ContractId   | 1     | EMA-ID the token belongs to.                                                                                                                                                   |
| **printedNumber** | **string(150)**  | **?**     | **Might be used for manual authorisation.**                                                                                                                                        |
| expiryDate    | DateTimeType | 1     | Tokens may be used until the date of expiry is reached. To be handled by the partners systems. Expired roaming authorisations may be erased locally by each partner's systems. |

**Comment by simonphilips** [comment]
What does eMA-id mean? If it's something like a contract ID, do we actually need to store this at the CPO since we're going to set up some sort of short negotiation/authentication process before starting a session? As such, the exchanged RFID's would not be a whitelist but a 'recognition list' to find out which eMSP should be contacted for the above negotation process.

**Comment by simonschllng** [comment]
eMA-ID is eMI3-wording. This is the contract ID but in the new format (2+3+C+8+Checksum)

**Comment by simonphilips** [comment]
I see, thanks. Then it seems that this is more a clearing-house concept that doesn't really have a meaning in a direct protocol such as OCPI, especially when we implement this authentication process. Is there a reason why we would want such information in the CPO, and where does it come from if there is no central system that keeps contracts?

**Comment by rutgerplantenga** [comment]
It's the Contract ID as allocated by the Service Provider (not related to
central systems). It's relevant for support purposes as the Contract ID (or
printed ID) is the only thing the customer can refer to when calling the
Operator's service desk. It is then very practical if the Operator's
support team is able to lookup that ContractID/PrintedID to see that it
belongs to a certain provider, to see what RFID/UID it relates to, etc.


*Rutger Plantenga*
+316 2907 6069

Keizersgracht 585
1017 DR Amsterdam

Postbus 3966
1001 AT Amsterdam

**Comment by simonphilips** [comment]
Aha, I was thinking about another type of contract (CPO-eMSP contract in a clearing house). So basically it's a visual card id? Can't we then simply use one field "printedNumber" instead? If available, you use the value of the eMA-ID, and otherwise you use some other unique number printed on the card? I'm trying to avoid having two fields with the same purpose because then it can become a bit confusing for everyone involved.

**Comment by simonphilips** [comment]
> and otherwise you use some other unique number printed on the card?

Actually, I should further clarify this: I meant unique for that eMSP only. Our printed card numbers are currently just numbers that are unique for us, but if another eMSP also uses numbers, we'll run into a conflict. At that point the CPO will have to ask the driver for the eMSP's name as well. Perhaps similar to the EVSE's, we will also need an optional (sub)provider field to solve that issue.

**Comment by cristinarc** [comment]
To be discussed later on, take into consideration for the 4th release. Cause some of the involved companies have CIR and don't require this.

**Comment by simonschllng** [comment]
There are three possible scenarios:
* The card has a visual globally unique number: _The services desk can find the number immediately._
* The card has a visual globally unique contract id but the contract has multiple cards. _Therefore the card has also another visual card number. The service desk would need both to identify the card._
* The card has a visual number that is unique to the eMSP only. _This is common for all companies which are not pure eMSPs but have an existing card system. The user would name the eMSP name and the cards number. The service desk can identify the card._

If we provide a contract_id field and a printed_number field, everything should be covered.
So far I'm on the same page as @simonphilips. However the (sub)provider-concept is not clear to me.

**Comment by RobertdeLeeuw-IHomer** [comment]
Is part of OCPI 2.0
